doc/sphinx/notebooks/advanced/varying_noise.pct.py

```diff
@@ -24,284 +24,314 @@
 )
+print_summary(model)
 
 # %%
-# model construction (notice that num_latent_gps is 1)
-kernel = gpflow.kernels.Matern52(lengthscales=0.5)
-model = gpflow.models.VGP((X, Y_data), kernel=kernel, likelihood=likelihood, num_latent_gps=1)
+mean, var = model.predict_y(X_plot_and_group)
+plot_distribution(X, Y, X_plot, mean.numpy(), var.numpy())
 
+# %% [markdown]
+# ## Demo 3: Empirical noise variance
+#
+# In this demo we will assume that you have multiple measurements at each $x$ location, and want to use the empirical variance at each location.
+
+# %% [markdown]
+# ### Generate data
 
 # %%
-for _ in range(reduce_in_tests(1000)):
-    natgrad.minimize(model.training_loss, [(model.q_mu, model.q_sqrt)])
+n_data = reduce_in_tests(20)
+n_repeats = reduce_in_tests(10)
+
+
+def generate_data():
+    rng = np.random.default_rng(42)  # for reproducibility
+    X = rng.uniform(0.0, 1.0, (n_data, 1))
+    signal = np.sin(6 * X)
+    noise_scale = rng.uniform(0.1, 1.0, (n_data, 1))
+    noise = noise_scale * rng.standard_normal((n_data, n_repeats))
+    Y = signal + noise
+    return X, Y
+
+
+X, Y = generate_data()
+Y_mean = np.mean(Y, axis=-1, keepdims=True)
+Y_var = np.var(Y, axis=-1, keepdims=True)
 
 # %% [markdown]
-# We've now fitted the VGP model to the data, but without optimizing over the hyperparameters. Plotting the data, we see that the fit is not terrible, but hasn't made use of our knowledge of the varying noise.
+# For the sake of plotting we'll create flat lists of (x, y) pairs:
 
 # %%
-# let's do some plotting!
-xx = np.linspace(-5, 5, 200)[:, None]
+X_flat = np.broadcast_to(X, Y.shape).reshape((-1, 1))
+Y_flat = Y.reshape((-1, 1))
+plot_distribution(X_flat, Y_flat)
 
-mu, var = model.predict_f(xx)
-
-fig, ax = plt.subplots(1, 1, figsize=(12, 6))
-ax.plot(xx, mu, "C0")
-ax.plot(xx, mu + 2 * np.sqrt(var), "C0", lw=0.5)
-ax.plot(xx, mu - 2 * np.sqrt(var), "C0", lw=0.5)
+# %% [markdown]
+# ### Data structure
+#
+# Our GPs don't like it when multiple data points occupy the same $x$ location. So we'll reduce this dataset to the means and the variances at each $x$, and use a custom function to model inject the empirical variances into the model.
 
-ax.plot(X, Y, "C1x", mew=2)
-_ = ax.set_xlim(-5, 5)
+# %%
+Y_mean = np.mean(Y, axis=-1, keepdims=True)
 
 # %% [markdown]
-# ### Optimizing the noise variances
-# Here we'll optimize over both the noise variance and the variational parameters, applying natural gradients interleaved with the Adam optimizer.
+# ### Fit a naive model
+#
+# Again we'll start by fitting a naive GPR, to demonstrate the problem.
 #
-# As before, we mark the variational parameters as not trainable, in order to train them separately with the natural gradient method.
+# Notice:
+# 1. We cannot actually model the point-wise variance. We just model `Y_mean`.
+# 2. Similarly we cannot actually plot the noise variance. In the other section we were plotting `predict_y` - here we plot `predict_x`.
```

Comment:
Why? Clearly it is going to be wrong as you have replaced the real data with it's average, but maybe this needs rephrasing as you should still be able to call `predict_y`?